src/app/components/Ad/index.tsx

```diff
@@ -1,13 +1,30 @@
 import React, { useContext } from 'react';
 import useToggle from '#hooks/useToggle';
 import { RequestContext } from '#contexts/RequestContext';
+import crypto from 'crypto';
+import { ServiceContext } from '#app/contexts/ServiceContext';
 import AmpAd from './Amp';
 import CanonicalAd from './Canonical';
 import { AdProps } from './types';
 import AdBootstrapJs from './Canonical/AdBootstrapJs';
 
+const servicesForNonce = ['mundo', 'arabic'];
+const countriesForNonce = ['eg', 'mx'];
+
 const AdContainer = ({ slotType, className, adcampaign }: AdProps) => {
-  const { isAmp, isLite, showAdsBasedOnLocation } = useContext(RequestContext);
+  const { isAmp, isLite, showAdsBasedOnLocation, country } =
+    useContext(RequestContext);
+
+  const { service } = useContext(ServiceContext);
+  const isNonceAllowed =
+    servicesForNonce.includes(service) &&
+    country &&
+    countriesForNonce.includes(country);
+
+  const nonce = isNonceAllowed
+    ? crypto.randomBytes(16).toString('hex')
```

Comment:
We need to add a prefix
```suggestion
    ? `nonce-${crypto.randomBytes(16).toString('hex')}`
```

Reply:
I think this value will be used as a `nonce` attribute value on the `script`, so it can just be a random string. However, when `nonce` is added to the CSP, then it should definitely [include the `nonce` prefix](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Global_attributes/nonce#:~:text=Sending%20a%20nonce%20with%20a%20CSP%20header)

Since this `nonce` won’t be used in the CSP yet, perhaps we could use `window.crypto.randomUUID()` instead of `crypto`, so the crypto module isn’t added to the bundle? [This is also suggested by the Copilot](https://github.com/bbc/simorgh/pull/12892#:~:text=src/app/components/Ad/index.tsx%3A25)

Reply:
Ah OK. Let's leave the `nonce-` prefix until we need it in the CSP, and yes, I agree - we don't need the `crypto` library - there's a helper we can use instead: https://github.com/bbc/simorgh/blob/latest/src/app/lib/utilities/getUUID/index.ts

Reply:
Can we please use the getUUID utility?